cmake/GetTargetArch.cmake

```diff
@@ -0,0 +1,16 @@
+function(get_target_arch FLAG)
+  execute_process(
+    COMMAND ${CMAKE_CXX_COMPILER} ${CMAKE_CXX_FLAGS} -dumpmachine
```

Comment:
This just doesn't work on Windows.

Reply:
I didn't mean to run that code on windows, i moved it down to after the if(UNIX)